File: Easy/2540_Minimum_Common_Value/Solution.py
from typing import List


class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        i = 0
        j = 0

        while (i < len(nums1) and j < len(nums2)):
            if (nums1[i] == nums2[j]):
                return nums2[j]
            elif (nums1[i] < nums2[j]):
                i += 1
            else:
                j += 1
        
        return -1


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    nums1 = [1, 2, 3]
    nums2 = [2, 4]
    print(solution.getCommon(nums1, nums2))

    # test cases 2
    nums1 = [1, 2, 3, 6]
    nums2 = [2, 3, 4, 5]
    print(solution.getCommon(nums1, nums2))
    
    

File: Easy/2540_Minimum_Common_Value/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.




class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int i = 0, j = 0;
        
        while (i < nums1.size() && j < nums2.size()) {
            if (nums1[i] == nums2[j]) {
                return nums1[i];
            }
            if (nums1[i] < nums2[j]) {
                i++;
            } else {
                j++;
            }
        }

        return -1;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;
    
    vector<int> nums11 = {1,2,3}, nums12 = {2,4};
    cout << solution.getCommon(nums11, nums12) << endl;

    vector<int> nums21 = {1,2,3,6}, nums22 = {2,3,4,5};
    cout << solution.getCommon(nums21, nums22) << endl;

    return 0;
}